### ISSUE TITLE
Type printing does not detect that a name is shadowed in the current scope

### ISSUE THREAD

**Issue by dragomirtitian** [microsoft/TypeScript#58456]
### 🔎 Search Terms

global shadowed declarations 

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed

### ⏯ Playground Link

[Playground Link](https://www.typescriptlang.org/dev/bug-workbench/?target=7&ts=5.5.0-dev.20240507#code/PTAEAEGcBcCcEsDG0Bco4FcCmAoEEATLRAGwENYzp4B7AOzU133GgoHMtVQtIAmAAyCceMOABm8EljpkAtljRkAdNEg4sADwAONWNFCJ6MUDVABeUAG8A2gGUAnnIBGNEsvjQslaHoC6aACMAL4iLJLSsgpozqrq8HK6+tamoMGg4rA0cqAARMrAZLkA3Bo6egakZJCQoI4ubimhWkmVxgbsFqbFQA)

### 💻 Code

```ts
// @strict: true
// @declaration: true
// @target: es2020

// @filename: a.ts
export const o = {[Symbol.iterator]: 1}

// @filename: b.ts

import { o } from "./a";
export class Symbol { }
export const g = o;
```

### 🙁 Actual behavior

The typeof of `g` is `{ [Symbol.iterator]: number; }`. This leads to an incorrect declaration since `Symbol` is shadowed in the current scope

### 🙂 Expected behavior

The typeof of `g` is `{ [globalThis.Symbol.iterator]: number; }`.
